dpnp/dpnp_algo/dpnp_elementwise_common.py

```diff
@@ -43,9 +43,432 @@
+        **kwargs,
+    ):
+        dpnp.check_supported_arrays_type(
+            *args, scalar_type=True, all_scalars=False
+        )
+        if kwargs:
+            raise NotImplementedError(
+                f"Requested function={self.name_} with kwargs={kwargs} "
+                "isn't currently supported."
+            )
+        if where is not True:
+            raise NotImplementedError(
+                f"Requested function={self.name_} with where={where} "
+                "isn't currently supported."
+            )
+        if subok is not True:
+            raise NotImplementedError(
+                f"Requested function={self.name_} with subok={subok} "
+                "isn't currently supported."
+            )
+        if dtype is not None and out is not None:
+            raise TypeError(
+                f"Requested function={self.name_} only takes `out` or `dtype`"
+                "as an argument, but both were provided."
+            )
+        if order is None:
+            order = "K"
+        elif order in "afkcAFKC":
+            order = order.upper()
+        else:
+            raise ValueError(
+                f"order must be one of 'C', 'F', 'A', or 'K' (got '{order}')"
+            )
+
+        if self.nin_ == 1:
+            if out is None and len(args) == 2:
+                out = args[1]
+            x = args[0]
+            if dtype is not None:
+                x = dpnp.astype(x, dtype=dtype, casting=casting, copy=False)
+            x_usm = dpnp.get_usm_ndarray(x)
+            astype_usm_args = (x_usm,)
+        else:
+            if out is None and len(args) == 3:
+                out = args[2]
+            x1, x2 = args[0], args[1]
+            if dtype is not None:
+                if dpnp.isscalar(x1):
+                    x1 = dpnp.asarray(x1, dtype=dtype)
+                    x2 = dpnp.astype(
+                        x2, dtype=dtype, casting=casting, copy=False
+                    )
+                elif dpnp.isscalar(x2):
+                    x1 = dpnp.astype(
+                        x1, dtype=dtype, casting=casting, copy=False
+                    )
+                    x2 = dpnp.asarray(x2, dtype=dtype)
+                else:
+                    x1 = dpnp.astype(
+                        x1, dtype=dtype, casting=casting, copy=False
+                    )
+                    x2 = dpnp.astype(
+                        x2, dtype=dtype, casting=casting, copy=False
+                    )
+
+            x1_usm = dpnp.get_usm_ndarray_or_scalar(x1)
+            x2_usm = dpnp.get_usm_ndarray_or_scalar(x2)
+            astype_usm_args = x1_usm, x2_usm
+
+        out_usm = None if out is None else dpnp.get_usm_ndarray(out)
+
+        res_usm = self.func.__call__(*astype_usm_args, out=out_usm, order=order)
+
+        if out is not None and isinstance(out, dpnp_array):
+            return out
+        return dpnp_array._create_from_usm_ndarray(res_usm)
+
+    @property
+    def nin(self):
```

Comment:
We will need to add some tests to verify the attributes.  Also to enable some relating third party tests.